monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -689,68 +501,37 @@ def get_config_scan():
-                    issues_byte_array[ReportService.ISSUES_DICT.DRUPAL.value] = True
-                elif issue['type'] == 'zerologon':
-                    if not issue['password_restored']:
-                        issues_byte_array[ReportService.ISSUES_DICT.ZEROLOGON_PASSWORD_RESTORE_FAILED.value] = True
-                    issues_byte_array[ReportService.ISSUES_DICT.ZEROLOGON.value] = True
-                elif issue['type'].endswith('_password') and issue['password'] in config_passwords and \
-                        issue['username'] in config_users or issue['type'] == 'ssh':
-                    issues_byte_array[ReportService.ISSUES_DICT.WEAK_PASSWORD.value] = True
-                elif issue['type'] == 'strong_users_on_crit':
-                    issues_byte_array[ReportService.ISSUES_DICT.PTH_CRIT_SERVICES_ACCESS.value] = True
-                elif issue['type'].endswith('_pth') or issue['type'].endswith('_password'):
-                    issues_byte_array[ReportService.ISSUES_DICT.STOLEN_CREDS.value] = True
-
-        return issues_byte_array
-
-    @staticmethod
-    def get_warnings_overview(issues, cross_segment_issues):
-        warnings_byte_array = [False] * len(ReportService.WARNINGS_DICT)
+                # TODO check if this actually works, because stolen passwords get added to config
+                # so any password will be in config. We need to separate stolen passwords from initial
+                # passwords in config.
+                if ReportService._is_weak_credential_issue(issue, config_users, config_passwords):
+                    issue_set.add(ReportService.DerivedIssueEnum.WEAK_PASSWORD)
+                elif ReportService._is_stolen_credential_issue(issue):
+                    issue_set.add(ReportService.DerivedIssueEnum.STOLEN_CREDS)
 
-        for machine in issues:
-            for issue in issues[machine]:
-                if issue['type'] == 'island_cross_segment':
-                    warnings_byte_array[ReportService.WARNINGS_DICT.CROSS_SEGMENT.value] = True
-                elif issue['type'] == 'tunnel':
-                    warnings_byte_array[ReportService.WARNINGS_DICT.TUNNEL.value] = True
-                elif issue['type'] == 'shared_admins':
-                    warnings_byte_array[ReportService.WARNINGS_DICT.SHARED_LOCAL_ADMIN.value] = True
-                elif issue['type'] == 'shared_passwords':
-                    warnings_byte_array[ReportService.WARNINGS_DICT.SHARED_PASSWORDS.value] = True
+                issue_set.add(issue['type'])
 
-        if len(cross_segment_issues) != 0:
-            warnings_byte_array[ReportService.WARNINGS_DICT.CROSS_SEGMENT.value] = True
+        return issue_set
 
-        return warnings_byte_array
+    @staticmethod
+    def _is_weak_credential_issue(issue: dict, config_usernames: List[str], config_passwords: List[str]) -> bool:
+        # Only credential exploiter issues have 'credential_type'
+        return 'credential_type' in issue and \
```

Comment:
Wouldn't make more sense to clean this according to the type?

Reply:
What 'type'? You mean have a list of credential_type issues? I think that "if issue has credentials, then it's a credential issue" logic is good enough for this PR